### ISSUE TITLE
`Search` gives Error: Anomaly "Uncaught exception Not_found."

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#17963]
#### Description of the problem

```
File "./theories/bug_search_anom_not_found_03.v", line 176, characters 4-52:
Error: Anomaly "Uncaught exception Not_found."
Please report at http://coq.inria.fr/bugs/.
Raised at Int.Map.find in file "clib/int.ml", line 39, characters 14-29
Called from Evd.find in file "engine/evd.ml", line 664, characters 29-56
Called from Evd.expand_existential in file "engine/evd.ml", line 725, characters 21-35
Called from Namegen.visible_ids.visible_ids in file "engine/namegen.ml", line 314, characters 16-47
Called from CArray.Fun1.iter in file "clib/cArray.ml", line 664, characters 6-13
Called from CArray.Fun1.iter in file "clib/cArray.ml", line 664, characters 6-13
Called from CArray.Fun1.iter in file "clib/cArray.ml", line 664, characters 6-13
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 542, characters 20-25
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Constr.iter_with_binders in file "kernel/constr.ml", line 545, characters 17-22
Called from Namegen.visible_ids in file "engine/namegen.ml", line 321, characters 11-26
Called from Namegen.next_ident_away_for_default_printing in file "engine/namegen.ml", line 419, characters 16-39
Called from Namegen.compute_displayed_name_in_gen_poly in file "engine/namegen.ml", line 468, characters 19-65
Called from Namegen.compute_displayed_name_in in file "engine/namegen.ml" (inlined), line 471, characters 32-74
Called from Detyping.Avoid.compute_name in file "pretyping/detyping.ml", line 159, characters 9-73
Called from Detyping.detype_binder in file "pretyping/detyping.ml", line 1004, characters 9-77
Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 31, characters 17-27
Re-raised at CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 36, characters 4-11
Called from DAst.get in file "lib/dAst.ml" (inlined), line 27, characters 12-25
Called from Notation_ops.match_ in file "interp/notation_ops.ml", line 1418, characters 8-19
Called from Stdlib__List.fold_left2 in file "list.ml", line 153, characters 37-51
Called from Notation_ops.match_notation_constr in file "interp/notation_ops.ml", line 1671, characters 14-113
Called from Constrextern.extern_notation in file "interp/constrextern.ml", line 1250, characters 10-74
Called from Constrextern.extern in file "interp/constrextern.ml", line 909, characters 6-48
Called from CList.map_loop in file "clib/cList.ml", line 291, characters 21-24
Called from CList.map in file "clib/cList.ml", line 299, characters 4-18
Called from Constrextern.extern_notation in file "interp/constrextern.ml", line 1268, characters 20-271
Called from Constrextern.extern in file "interp/constrextern.ml", line 909, characters 6-48
Called from Printer.pr_letype_env in file "printing/printer.ml", line 93, characters 28-80
Called from Printer.pr_ltype_env in file "printing/printer.ml" (inlined), line 98, characters 2-74
Called from ComSearch.interp_search.pr_search in file "vernac/comSearch.ml", line 134, characters 17-63
Called from Search.iter_priority_queue in file "vernac/search.ml", line 159, characters 4-22
Called from ComSearch.interp_search in file "vernac/comSearch.ml", line 145, characters 6-139
Called from Vernacextend.vtreadproofopt.(fun) in file "vernac/vernacextend.ml", line 159, characters 29-44
Called from Vernacinterp.interp_typed_vernac in file "vernac/vernacinterp.ml", line 21, characters 20-113
Called from Vernacinterp.interp_control.(fun) in file "vernac/vernacinterp.ml", line 186, characters 24-113
Called from Flags.with_modified_ref in file "lib/flags.ml", line 17, characters 14-17
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Vernacinterp.interp_gen in file "vernac/vernacinterp.ml", line 230, characters 16-41
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Vernacinterp.interp in file "vernac/vernacinterp.ml", line 244, characters 15-72
Called from Stm.Reach.known_state.reach.(fun) in file "stm/stm.ml", line 2185, characters 16-43
Called from Stm.State.define in file "stm/stm.ml", line 965, characters 6-10
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Stm.Reach.known_state.reach in file "stm/stm.ml", line 2327, characters 4-105
Called from Stm.observe in file "stm/stm.ml", line 2424, characters 4-60
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Vernac.interp_vernac in file "toplevel/vernac.ml", line 79, characters 29-50
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Flags.with_modified_ref in file "lib/flags.ml", line 17, characters 14-17
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Util.try_finally in file "lib/util.ml", line 133, characters 16-19
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Util.try_finally in file "lib/util.ml", line 137, characters 6-32
Called from Vernac.load_vernac_core.loop in file "toplevel/vernac.ml", line 121, characters 8-352
Called from Vernac.load_vernac_core in file "toplevel/vernac.ml", line 134, characters 6-19
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Vernac.load_vernac in file "toplevel/vernac.ml", line 210, characters 30-83
Called from Ccompile.compile in file "toplevel/ccompile.ml", line 70, characters 18-78
Called from Ccompile.compile in file "toplevel/ccompile.ml" (inlined), line 126, characters 2-59
Called from Ccompile.compile_file in file "toplevel/ccompile.ml", line 135, characters 4-61
Called from Coqc.coqc_main in file "toplevel/coqc.ml", line 48, characters 2-100
Called from Coqc.coqc_run in file "toplevel/coqc.ml", line 67, characters 4-36
```

#### Coq Version

8.19+alpha 61ee398ed32f9334dd664ea8ed2697178e6e3844

**Comment by JasonGross** [comment]
@coqbot minimize coq.dev
```bash
git clone https://github.com/JasonGross/neural-net-coq-interp.git --branch=zzz-bug-anomaly-search
cd neural-net-coq-interp
make theories/bug_search_anom_not_found_03.vo -j2
```

**Comment by JasonGross** [comment]
@coqbot minimize coq.dev
```bash
git clone https://github.com/JasonGross/neural-net-coq-interp.git --branch=zzz-bug-anomaly-search
cd neural-net-coq-interp
make theories/bug_search_anom_not_found_03.vo -j2
```

**Comment by SkySkimmer** [comment]
Unrelated Not_found

Ltac2 uses harcoded names instead of some Register-like mechanism so inlining the Ltac2 files won't work well.

**Comment by JasonGross** [comment]
@coqbot minimize coq.dev
```bash
git clone https://github.com/JasonGross/neural-net-coq-interp.git --branch=zzz-bug-anomaly-search
cd neural-net-coq-interp
make theories/bug_search_anom_not_found_03.vo -j2
```

**Comment by herbelin** [comment]
In the minimized example, the problem is caused by an existential variable in the hypothesis `H` of the goal.

An alternative shorter example, not failing though, is:
```
Goal exists y, Some y = Some y :> option nat -> True.
eexists. intro H.
Search Some eq. 
```
which returns
```
H: Some ?X8 = Some ?X8
```
instead of
```
H: Some ?y = Some ?y
```
Or also:
```
Goal exists y, Some y = Some y :> option nat -> True.
eexists ?[y]. [y]:exact 0. intro H.
Search Some eq. 
```
which shows `H: Some ?X8 = Some ?X8` instead of `H: Some 0 = Some 0`.

Fixed in #17987.
